blockchain/blockchain.go

```diff
@@ -715,6 +715,12 @@ func (b *Blockchain) StateAtBlockNumber(blockNumber uint64) (core.StateReader, S
 
 // StateAtBlockHash returns a StateReader that provides a stable view to the state at the given block hash
 func (b *Blockchain) StateAtBlockHash(blockHash *felt.Felt) (core.StateReader, StateCloser, error) {
+	if blockHash.IsZero() {
```

Comment:
Maybe the caller should create the empty state?

Reply:
Hm.. not sure, we would implement it over and over again on caller side I guess. @omerfirmak wdyt?

Reply:
> the caller should create the empty state

I think it fits well in the API of `blockchain.Reader`, we can keep it here.
We can make `emptyState` a singleton tho, to avoid instantiating it every time. But it's not a priority either.

Reply:
> We can make emptyState a singleton tho, to avoid instantiating it every time. But it's not a priority either.

Good point I'll add it to next PR